### ISSUE TITLE
Created Type does not keep it's Z value when allocating to a Building

### ISSUE THREAD

**Issue by Ave-5D** [IfcOpenShell/IfcOpenShell#4186]
[3bd hous240109.txt](https://github.com/IfcOpenShell/IfcOpenShell/files/13874657/3bd.hous240109.txt)
![Skermskoot (83)](https://github.com/IfcOpenShell/IfcOpenShell/assets/36395977/05deef7a-739f-42b0-9476-762c36b7add2)

**Comment by theoryshaw** [comment]
I didn't totally follow you, but does this help?...

https://github.com/IfcOpenShell/IfcOpenShell/assets/507113/acdea6d9-a223-4962-a296-aabb4ce9dc9d

**Comment by ppaawweeuu** [comment]
It seems that new object positioning depends on active Collection.
If we activate IfcProject, IfcSite, IfcStorey, IfcSpace Collection - bottom of added object boundary box (not origin point) is glued to spatial container z-value (project, site, building or storey Empty object). x and y values are from Cursor position.
If we activate any different Collection - origin point of added object is glued to Cursor location (x,y,z).
Active collection is just recently selected collection. Its icon is slightly highlighted.
@Ave-5D - can you test it that way?

![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/61344631/881b83f2-5238-459d-8102-57d622414048)

**Comment by Ave-5D** [comment]
Hi PPaawweeuu, when the Collection (IFCSite/My Site) is not active then the Z value is as per the Type value and then it needs to be moved into Collection (IFCSite/MySite), when Collection (IFCSite/My Site) is active then the Z value is lost.

 

Ryan’s video works because the added element is not in the Collection (IFCSite/My Site).

 

Thanks now I know what I need to do the keep the Type Z values.

 

Kind regards.

**Comment by Moult** [comment]
FYI the added occurrence's name is not "lost", the occurrence naming scheme differs from user to user and shouldn't be identical to the type's name in most cases. You can configure the occurrence naming scheme in the addon preferences.

**Comment by Ave-5D** [comment]
Hi Dion, 

 

Thanks for the clarification, slowly but surely, I am learning, I changed my occurrence naming now to “By Type”.

I can’t see an option for “PredifinedType/RelatingType”. I changed the addon preferences to “By Type”, I still get “SanitaryTerminal”(Type) as the default name and not “ToiletPan”(PredifindedType/RelatingType).

 

I fully agree that the occurrence name can’t be identical to the Type’s name. That the occurrence name can be edited by the user is critical. 

 

As a user it would be simple not to manually change the occurrence name to the PredifinedType/RelatingType +unique number but that BlenderBIM occurrence name would be RelatingType+unique number.

 

Once again thanks for your effort.

 

Kind regards

 

Roel
